go/arrow/schema_test.go

```diff
@@ -99,6 +100,9 @@ func TestMetadata(t *testing.T) {
 			if got, want := tc.md.String(), tc.serialize; got != want {
 				t.Fatalf("invalid stringer: got=%q, want=%q", got, want)
 			}
+			if len(tc.kvs) != 0 {
+				assert.Equal(t, tc.kvs, md.ToMap())
+			}
```

Comment:
how about adding a round trip test:

create a map, construct the metadata from the map, then call `ToMap` on the result and assert the two maps are equal.

Reply:
I think we already have a round trip test here. We have `MetadataFrom` call on line 81

Reply:
ah perfect